src/utils/searchHelpers.ts

```diff
@@ -0,0 +1,29 @@
+interface PropertySearch {
+  key: string;
+  value: string | string[]; // Can check for one of multiple values
+}
+interface SearchConfig {
+  searchTerm?: string;
+  properties?: PropertySearch[];
+  canMatchAnyProperty?: boolean; // Whether to use "OR" or "AND" operator for each property check
+  canMatchAnySubquery?: boolean; // Whether to use "OR" or "AND" operator for each subquery
+}
```

Comment:
`useSearchRegistration` burde kanskje ta inn `searchConfig` i stedet for `searchTerm`, trenger vi bare kalle `createSearchQuery` derfra

Reply:
https://github.com/BIBSYSDEV/NVA-Frontend/pull/1362